File: rs/005.rs
fn gcd(a: i64, b: i64) -> i64 {
    if b == 0 {
        a
    } else {
        gcd(b, a % b)
    }
}

fn lcm(a: i64, b: i64) -> i64 {
    a * b / gcd(a, b)
}

fn main() {
    let mut n = 1;
    for i in 1..=20 {
        n = lcm(n, i);
    }
    println!("{}", n);
}

File: go/005/005.go
package main

import "fmt"

func main() {
	n := 1
	for i := 1; i <= 20; i++ {
		n = lcm(n, i)
	}
	fmt.Println(n)
}

func lcm(a, b int) int {
	return a * b / gcd(a, b)
}

func gcd(a, b int) int {
	if b == 0 {
		return a
	}
	return gcd(b, a % b)
}

